[addons] add next step on CAddonInfo to become independent

AlwinEsch opened the pull request:
## Description
<!--- Describe your change in detail -->
This change all values on CAddonInfo to private and access
them over functions.

Main reason for them is to make the class more safe for next changes
where CAddonInfo is no longer in CAddon, but globally as replacement
for cpluff.

Set of this value becomes in future mainly done from the class constructor
and no more by external parts.

<!--- Provide a general summary of your change in the Title above -->


## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

ksooo reviewed (commented):
Some suggestions and questions...

tamland commented on xbmc/addons/Addon.h:
```diff
@@ -56,38 +56,38 @@ class CAddon : public IAddon
   virtual ~CAddon() {}
 
-  TYPE Type() const override { return m_addonInfo.type; }
+  TYPE Type() const override { return m_addonInfo.MainType(); }
```
In the end though there really should be no such thing as an 'addon type', but only addons with N possible extension points (which can have a type). So assuming this is only temporary for legacy code I guess 'MainType' works as to making it clearer..

tamland commented on xbmc/addons/AddonInfo.h:
```diff
@@ -85,13 +85,240 @@ namespace ADDON
+     * @note This function is only be related to constructor's who read the
+     * addon xml part.
+     */
+    bool IsUsable() const { return m_usable; }
```
This doesn't belong to this commit

AlwinEsch replied:
Can remove it now, but becomes on next commits used the know class construction with read of addon.xml was OK.

tamland commented on xbmc/addons/AddonInfo.h:
```diff
@@ -85,13 +85,240 @@ namespace ADDON
+    void SetLastUpdated(CDateTime lastUpdated) { m_lastUpdated = lastUpdated; }
+    void SetLastUsed(CDateTime lastUsed) { m_lastUsed = lastUsed; }
+    void SetOrigin(std::string origin) { m_origin = std::move(origin); }
+    //@}
```
Same here: not used anywhere. And definitely not used by CAddonDatabase.

AlwinEsch replied:
Also a part who becomes in future used from CAddonDatabase. Can remove it and add later.

tamland commented on xbmc/addons/AddonInfo.h:
```diff
@@ -85,13 +85,240 @@ namespace ADDON
+    /*!
+     * @brief Get the ID from add-on
+     *
+     * The ID is defined in addon.xml but should also equal with add-on path!
```
This isn't correct. There is no such requirement.

tamland commented on xbmc/addons/AddonInfo.h:
```diff
@@ -85,13 +85,240 @@ namespace ADDON
+     * @brief Licence text of add-on
+     *
+     * @return the complete licence text defined on addon.xml
+     */
```
I find all these getter 'docstrings' very superficial.. It's completely obvious to everyone what all these methods does.

AlwinEsch replied:
Better remove the docs, for me also not really needed?

Thought only that maybe wanted for there.

ksooo replied:
What @tamland said. Remove it. Does  not really add any value.

tamland commented on xbmc/addons/AddonInfo.h:
```diff
@@ -85,13 +85,240 @@ namespace ADDON
+    /*!
+     * @brief To get install date of add-on
+     *
+     * @note this is only be set from add-on browser and not always present!
```
That's not true. It should be set everywhere except when getting a single addon from addon manager. (And that is just a workaround due to some parts calling GetAddon way too much)

AlwinEsch commented on xbmc/addons/AddonBuilder.h:
```diff
@@ -30,36 +30,36 @@ class CAddonBuilder
-  const std::string& GetId() const { return m_addonInfo.id; }
-  const AddonVersion& GetVersion() const { return m_addonInfo.version; }
+  const std::string& GetId() const { return m_addonInfo.m_id; }
+  const AddonVersion& GetVersion() const { return m_addonInfo.m_version; }
```
The add of `friend class CAddonBuilder` is only temporarily to allow them to set the values in CAddonInfo, on end create CAddonInfo the values itself by his constructor.

ksooo replied:
Okay.

AlwinEsch commented on xbmc/addons/AddonInfo.h:
```diff
@@ -101,31 +328,40 @@ namespace ADDON
+    InfoMap extrainfo; // defined here, but removed in future with cpluff
+
+  private:
+    friend class ADDON::CAddonBuilder;
```
CAddonBuilder is the only class who currently need to set the values on CAddonInfo and only temporarily, the builder becomes on next steps complete removed. See also my other comment,

ksooo replied:
Okay, then.

ksooo commented:
Let's handle documentation issues in a separate PR, once all code changes are in, like agreed before.

AlwinEsch committed:
[addons] add next step on CAddonInfo to become independent

This change all values on CAddonInfo to private and access
them over functions.

Main reason for them is to make the class more safe for next changes
where CAddonInfo is no longer in CAddon, but globally as replacement
for cpluff.

Set of this value becomes in future mainly done from the class constructor
and no more by external parts.

AlwinEsch commented:
Have removed the comments, but want to leave the "Parts used from ADDON::CAddonDatabase" in. Becomes used on next commits and want them to hold small to see the critical parts.

ksooo commented:
Fine by me. Merge it!

AlwinEsch commented:
jenkins build this please
